quickwit/quickwit-ui/src/components/TimeRangeSelect.tsx

```diff
@@ -282,29 +251,19 @@ function CustomDatesPanel(props: SearchComponentProps): JSX.Element {
 interface DateTimeRangeLabelProps {
   startTimestamp: number | null;
   endTimestamp: number | null;
-  timeUnit: TimeUnit;
 }
 
 function DateTimeRangeLabel(props: DateTimeRangeLabelProps): JSX.Element {
-  const [startTimestamp, setStartTimestamp] = useState(convertToMilliSecond(props.startTimestamp, props.timeUnit));
-  const [endTimestamp, setEndTimestamp] = useState(convertToMilliSecond(props.endTimestamp, props.timeUnit));
-  const dateTimeFormat = getDateTimeFormat(props.timeUnit);
-
-  useEffect(() => {
-    setStartTimestamp(convertToMilliSecond(props.startTimestamp, props.timeUnit));
-    setEndTimestamp(convertToMilliSecond(props.endTimestamp, props.timeUnit));
-  }, [props.startTimestamp, props.endTimestamp, props.timeUnit])
-
   function Label() {
-    if (startTimestamp !== null && endTimestamp !== null) {
+    if (props.startTimestamp !== null && props.endTimestamp !== null) {
       return <>
-        {dayjs(startTimestamp).utc().format(dateTimeFormat)} -{" "}
-        {dayjs(endTimestamp).utc().format(dateTimeFormat)}
+        {dayjs(props.startTimestamp * 1000).utc().format(DATE_TIME_WITH_SECONDS_FORMAT)} -{" "}
+        {dayjs(props.endTimestamp * 1000).utc().format(DATE_TIME_WITH_SECONDS_FORMAT)}
       </>
-    } else if (startTimestamp !== null && endTimestamp === null) {
-      return <>Since {dayjs(startTimestamp).utc().fromNow(true)}</>
-    } else if (startTimestamp == null && endTimestamp != null) {
-      return <>Before {dayjs(endTimestamp).utc().format(dateTimeFormat)}</>
+    } else if (props.startTimestamp !== null && props.endTimestamp === null) {
+      return <>Since {dayjs(props.startTimestamp * 1000).utc().fromNow(true)}</>
+    } else if (props.startTimestamp == null && props.endTimestamp != null) {
+      return <>Before {dayjs(props.endTimestamp * 1000).utc().format(DATE_TIME_WITH_SECONDS_FORMAT)}</>
```

Comment:
Since we're getting the milliseconds of the timestamp I assume, let's extract this out to be a separate method so that way we're not worrying about each timestamp being converted correctly and consistently.

Reply:
Oh man, I was so lazy on this. Let's fix that.